<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Alexandru T. Codilean" />

<meta name="date" content="2025-02-18" />

<title>Case Study: Koala habitat suitability</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #204a87; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #204a87; font-weight: bold; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #ce5c00; font-weight: bold; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SCII205</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fab fa-r-project"></span>
     
    Computer Labs
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="r1.html">Introduction to R</a>
    </li>
    <li>
      <a href="r2.html">Spatial data in R (part 1)</a>
    </li>
    <li>
      <a href="r3.html">Spatial data in R (part 2)</a>
    </li>
    <li>
      <a href="r4.html">Case Study 1: Rainfed crops</a>
    </li>
    <li>
      <a href="r5.html">Case Study 2: Koala habitat</a>
    </li>
  </ul>
</li>
<li>
  <a href="lectnotes.html">
    <span class="fa fa-rectangle-list"></span>
     
    Lecture Notes
  </a>
</li>
<li>
  <a href="downloads.html">
    <span class="fa fa-cloud-arrow-down"></span>
     
    Downloads
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Case Study: Koala habitat suitability</h1>
<h3 class="subtitle">SCII205: Introductory Geospatial Analysis /
Practical No. 5</h3>
<h4 class="author">Alexandru T. Codilean</h4>
<h4 class="date">2025-02-18</h4>

</div>


<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p><img src="img/ecological_report.png" width="25%" style="float:left; margin-right:1em;" /></p>
<p>In this practical, we will replicate a koala population
sustainability study conducted in the Lower Hunter region of NSW for the
<em>Department of Sustainability, Environment, Water, Population and
Communities</em> by the consultancy <em>Eco Logical</em>.</p>
<p>This study serves as an excellent example of applying geospatial
analysis to discrete entities. It involves intersecting multiple
thematic data layers, each assigned weights based on attribute values,
feature geometry, or proximity to other layers. To ensure the practical
is manageable within the three-hour timeframe, some layers have been
pre-prepared. Additionally, the study area has been reduced to three of
the five Lower Hunter LGAs analyzed in the original Eco Logical
study.</p>
<p>The goal of this practical is to prepare the remaining data layers
and conduct the final overlay operation to generate a koala habitat
value index.</p>
<p>Further details can be found in the <em>Eco Logical</em> Final
Report, which is included in this practical’s data package and available
for download from the Australian Government <a
href="https://www.dcceew.gov.au/sites/default/files/env/pages/25570c73-a276-4efb-82f4-16f802320e62/files/ecological-koala.pdf">website</a>.</p>
<p><font size="3"><svg viewBox="0 0 512 512" style="height:1em;position:relative;display:inline-block;top:.1em;" xmlns="http://www.w3.org/2000/svg">
<path d="M436.67,184.11a27.17,27.17,0,0,1-38.3,0l-22.48-22.49a27.15,27.15,0,0,1,0-38.29l50.89-50.89a.85.85,0,0,0-.26-1.38C393.68,57,351.09,64.15,324.05,91c-25.88,25.69-27.35,64.27-17.87,98a27,27,0,0,1-7.67,27.14l-173,160.76a40.76,40.76,0,1,0,57.57,57.54l162.15-173.3A27,27,0,0,1,372,253.44c33.46,8.94,71.49,7.26,97.07-17.94,27.49-27.08,33.42-74.94,20.1-102.33a.85.85,0,0,0-1.36-.22Z" style="fill:none;stroke:#000;stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"></path>
<path d="M224,284c-17.48-17-25.49-24.91-31-30.29a18.24,18.24,0,0,1-3.33-21.35,20.76,20.76,0,0,1,3.5-4.62l15.68-15.29a18.66,18.66,0,0,1,5.63-3.87,18.11,18.11,0,0,1,20,3.62c5.45,5.29,15.43,15,33.41,32.52" style="fill:none;stroke:#000;stroke-linecap:round;stroke-linejoin:round;stroke-width:32px"></path>
<path d="M317.07,291.3c40.95,38.1,90.62,83.27,110,99.41a13.46,13.46,0,0,1,.94,19.92L394.63,444a14,14,0,0,1-20.29-.76c-16.53-19.18-61.09-67.11-99.27-107" style="fill:none;stroke:#000;stroke-linecap:round;stroke-linejoin:round;stroke-width:32px"></path>
<path d="M17.34,193.5l29.41-28.74a4.71,4.71,0,0,1,3.41-1.35,4.85,4.85,0,0,1,3.41,1.35h0a9.86,9.86,0,0,0,8.19,2.77c3.83-.42,7.92-1.6,10.57-4.12,6-5.8-.94-17.23,4.34-24.54a207,207,0,0,1,19.78-22.6c6-5.88,29.84-28.32,69.9-44.45A107.31,107.31,0,0,1,206.67,64c22.59,0,40,10,46.26,15.67a89.54,89.54,0,0,1,10.28,11.64A78.92,78.92,0,0,0,254,88.54,68.82,68.82,0,0,0,234,87.28c-13.33,1.09-29.41,7.26-38,14-13.9,11-19.87,25.72-20.81,44.71-.68,14.12,2.72,22.1,36.1,55.49a6.6,6.6,0,0,1-.34,9.16l-18.22,18a6.88,6.88,0,0,1-9.54.09c-21.94-21.94-36.65-33.09-45-38.16s-15.07-6.5-18.3-6.85a30.85,30.85,0,0,0-18.27,3.87,11.39,11.39,0,0,0-2.64,2,14.14,14.14,0,0,0,.42,20.08l1.71,1.6a4.63,4.63,0,0,1,0,6.64L71.73,246.6A4.71,4.71,0,0,1,68.32,248a4.86,4.86,0,0,1-3.41-1.35L17.34,200.22A4.88,4.88,0,0,1,17.34,193.5Z" style="fill:none;stroke:#000;stroke-linecap:round;stroke-linejoin:round;stroke-width:32px"></path></svg>
<strong>R packages</strong></font></p>
<p>The practical will primarily utilize R functions from packages
explored in previous sessions:</p>
<ol style="list-style-type: decimal">
<li><p>The <strong>tidyverse</strong>: is a collection of R packages
designed for data science, providing a cohesive set of tools for data
manipulation, visualization, and analysis. It includes core packages
such as <strong>ggplot2</strong> for visualization,
<strong>dplyr</strong> for data manipulation, <strong>tidyr</strong> for
data tidying, and <strong>readr</strong> for data import, all following
a consistent and intuitive syntax.</p></li>
<li><p>The <strong>sf</strong> package: provides a standardized
framework for handling spatial vector data in R based on the simple
features standard (OGC). It enables users to easily work with points,
lines, and polygons by integrating robust geospatial libraries such as
GDAL, GEOS, and PROJ.</p></li>
<li><p>The <strong>terra</strong> package: is designed for efficient
spatial data analysis, with a particular emphasis on raster data
processing and vector data support. As a modern successor to the
<strong>raster</strong> package, it utilizes improved performance
through C++ implementations to manage large datasets and offers a suite
of functions for data import, manipulation, and analysis.</p></li>
<li><p>The <strong>tmap</strong> package: provides a flexible and
powerful system for creating thematic maps in R, drawing on a layered
grammar of graphics approach similar to that of
<strong>ggplot2</strong>. It supports both static and interactive map
visualizations, making it easy to tailor maps to various presentation
and analysis needs.</p></li>
</ol>
<p><font size="3"><svg viewBox="0 0 512 512" style="height:1em;position:relative;display:inline-block;top:.1em;" xmlns="http://www.w3.org/2000/svg">
<path d="M256,160c16-63.16,76.43-95.41,208-96a15.94,15.94,0,0,1,16,16V368a16,16,0,0,1-16,16c-128,0-177.45,25.81-208,64-30.37-38-80-64-208-64-9.88,0-16-8.05-16-17.93V80A15.94,15.94,0,0,1,48,64C179.57,64.59,240,96.84,256,160Z" style="fill:none;stroke:#000;stroke-linecap:round;stroke-linejoin:round;stroke-width:32px"></path>
<line x1="256" y1="160" x2="256" y2="448" style="fill:none;stroke:#000;stroke-linecap:round;stroke-linejoin:round;stroke-width:32px"></line></svg>
<strong>Resources</strong></font></p>
<ul>
<li>The <em>Geocomputation with R</em> <a
href="https://r.geocompx.org">book</a>.</li>
<li>The <strong>tidyverse</strong> library of packages <a
href="https://www.tidyverse.org/learn/">documentation</a>.</li>
<li>The <strong>sf</strong> package <a
href="https://r-spatial.github.io/sf/">documentation</a>.</li>
<li>The <strong>terra</strong> package <a
href="https://rspatial.org/spatial/">documentation</a>.</li>
<li>The <strong>tmap</strong> package <a
href="https://r-tmap.github.io">documentation</a>.</li>
</ul>
</div>
<div id="study-methodology" class="section level1">
<h1>Study methodology</h1>
<p>The methodology behind the koala habitat study is described in detail
in the <em>Eco Logical</em> <a
href="https://www.dcceew.gov.au/sites/default/files/env/pages/25570c73-a276-4efb-82f4-16f802320e62/files/ecological-koala.pdf">Final
Report</a>. The following section reproduces information from Table 3 of
the report, providing information regarding the various data layers used
in the modelling done by <em>Eco Logical</em>.</p>
<div id="data-layers" class="section level2">
<h2>Data layers</h2>
<p>The data layers that are provided as part of this practical were
obtained from <em>Eco Logical</em> and represent the layers that were
used in their Final Report. The data layers that are derived as part of
the practical use raw data that is different to the data used by <em>Eco
Logical</em> in their study (e.g., roads, rail corridors, and built
areas are taken from a different source) and so the calculated final
layers may be different to the ones produced by <em>Eco
Logical</em>.</p>
<p>All data layers were cropped to the extent of the three Lower Hunter
LGAs considered here: Lake Macquarie, Newcastle, and Port Stephens.</p>
<div id="data-layers-provided" class="section level3">
<h3>Data layers provided</h3>
<p><img src="img/allveg_prim.png" width="20%" style="float:right; margin-left:1em;" /></p>
<div id="primary-feed-trees" class="section level4">
<h4>1. Primary feed trees</h4>
<ul>
<li><p><strong>Description:</strong><br />
Selection of vegetation types known to have primary preferred koala tree
species with the proportional abundance of the specific tree species
within the vegetation type identified as either Nil, low, moderate, or
high.</p></li>
<li><p><strong>Details:</strong><br />
An initial multiplier of <span class="math inline">\(\times 3\)</span>
was applied to these data to distinguish from secondary feed or
supplementary tree species.</p></li>
<li><p><strong>Values:</strong><br />
<code>0</code>: Outside of any vegetation classification containing
identified primary feed trees.<br />
<code>60</code>: Vegetation types with a low proportion of primary feed
trees.<br />
<code>150</code>: Vegetation types with a moderate proportion of primary
feed trees.<br />
<code>300</code>: Vegetation types with a high proportion of primary
feed trees.</p></li>
<li><p><strong>Weighting:</strong> <span class="math inline">\(\times
3\)</span></p></li>
</ul>
<hr />
<p><img src="img/allveg_sec.png" width="20%" style="float:right; margin-left:1em;" /></p>
</div>
<div id="secondary-feed-trees" class="section level4">
<h4>2. Secondary feed trees</h4>
<ul>
<li><p><strong>Description:</strong><br />
Selection of vegetation types known to have secondary preferred koala
tree species with the proportional abundance of the specific tree
species within the vegetation type identified as either Nil, low,
moderate, or high.</p></li>
<li><p><strong>Details:</strong><br />
An initial multiplier of <span class="math inline">\(\times 2\)</span>
was applied to these data to distinguish from primary feed or
supplementary tree species.</p></li>
<li><p><strong>Values:</strong><br />
<code>0</code>: Outside of any vegetation classification containing
identified secondary feed trees.<br />
<code>40</code>: Vegetation types with a low proportion of secondary
feed trees.<br />
<code>100</code>: Vegetation types with a moderate proportion of
secondary feed trees.<br />
<code>200</code>: Vegetation types with a high proportion of secondary
feed trees.</p></li>
<li><p><strong>Weighting:</strong> <span class="math inline">\(\times
2\)</span></p></li>
</ul>
<hr />
<p><img src="img/allveg_supp.png" width="20%" style="float:right; margin-left:1em;" /></p>
</div>
<div id="supplementary-trees" class="section level4">
<h4>3. Supplementary trees</h4>
<ul>
<li><p><strong>Description:</strong><br />
Selection of vegetation types known to have supplementary tree species
supporting shelter and roosting for koala, with the proportional
abundance of the specific tree species within the vegetation type
identified as either Nil, low, moderate, or high.</p></li>
<li><p><strong>Values:</strong><br />
<code>0</code>: Outside of any vegetation classification containing
identified supplementary trees.<br />
<code>20</code>: Vegetation types with a low proportion of supplementary
trees.<br />
<code>50</code>: Vegetation types with a moderate proportion of
supplementary trees.<br />
<code>100</code>: Vegetation types with a high proportion of
supplementary trees.</p></li>
<li><p><strong>Weighting:</strong> <span class="math inline">\(\times
1\)</span></p></li>
</ul>
<hr />
<p><img src="img/veg_waterprox.png" width="20%" style="float:right; margin-left:1em;" /></p>
</div>
<div id="vegetation-proximity-to-water" class="section level4">
<h4>4. Vegetation proximity to water</h4>
<ul>
<li><p><strong>Description:</strong><br />
Extant vegetation within 50 m of 3rd order or higher drainage
lines.</p></li>
<li><p><strong>Values:</strong><br />
<code>0</code>: Outside identified buffer.<br />
<code>100</code>: Within identified buffer.</p></li>
<li><p><strong>Weighting:</strong> <span class="math inline">\(\times
2\)</span></p></li>
</ul>
<hr />
<p><img src="img/patch.png" width="20%" style="float:right; margin-left:1em;" /></p>
</div>
<div id="vegetation-patch-size" class="section level4">
<h4>5. Vegetation patch size</h4>
<ul>
<li><p><strong>Description:</strong><br />
The size of a patch of vegetation. A patch is defined as an area of
consolidated vegetation that is separated from other patches by more
than 100 m by mapped infrastructure to represent consolidated koala
habitat.</p></li>
<li><p><strong>Values:</strong><br />
<code>0</code>: Smallest patch size.<br />
<code>100</code>: Patch at least 100 ha in size.</p></li>
<li><p><strong>Weighting:</strong> <span class="math inline">\(\times
3\)</span></p></li>
</ul>
<hr />
</div>
</div>
<div id="data-layers-derived-in-practical" class="section level3">
<h3>Data layers derived in practical</h3>
<p><img src="img/soilfert.png" width="20%" style="float:right; margin-left:1em;" /></p>
<div id="soil-fertility" class="section level4">
<h4>1. Soil fertility</h4>
<ul>
<li><p><strong>Description:</strong><br />
Identification of areas with high soil fertility.</p></li>
<li><p><strong>Values:</strong><br />
<code>0</code>: Outside of identified fertile soils.<br />
<code>100</code>: Within identified fertile soils.</p></li>
<li><p><strong>Weighting:</strong> <span class="math inline">\(\times
2\)</span></p></li>
</ul>
<hr />
<p><img src="img/koala.png" width="20%" style="float:right; margin-left:1em;" /></p>
</div>
<div id="buffered-koala-sightings" class="section level4">
<h4>2. Buffered koala sightings</h4>
<ul>
<li><p><strong>Description:</strong><br />
100 m buffer around known recorded sightings of koala.<br />
<strong>NB:</strong> This is not a comprehensive dataset; it is based
only on areas where sightings have been recorded.</p></li>
<li><p><strong>Values:</strong><br />
<code>0</code>: Not within 100 m of a known record.<br />
<code>75</code>: Within 100 m of a known record dated 1985 or
earlier.<br />
<code>100</code>: Within 100 m of a known record dated 1986 or
later.</p></li>
<li><p><strong>Weighting:</strong> <span class="math inline">\(\times
1\)</span></p></li>
</ul>
<hr />
<p><img src="img/dist_roads.png" width="20%" style="float:right; margin-left:1em;" /></p>
</div>
<div id="distance-to-infrastructure" class="section level4">
<h4>3. Distance to infrastructure</h4>
<ul>
<li><p><strong>Description:</strong><br />
300 m buffer around identified and mapped major barriers to koala
movement.</p></li>
<li><p><strong>Values:</strong><br />
<code>0</code>: Within 300 m of identified infrastructure.<br />
<code>100</code>: 300 m or greater from identified
infrastructure.</p></li>
<li><p><strong>Weighting:</strong> <span
class="math inline">\(\times3\)</span></p></li>
</ul>
<hr />
</div>
</div>
<div id="koala-habitat-value-index" class="section level3">
<h3>Koala habitat value index</h3>
<p>The final koala habitat suitability map is calculated by combining
all the scores assigned to each data layer (see <code>values</code>
above) with each layer further multiplied by a weight (see
<code>weighting</code> above).</p>
<p>The calculation is as follows:</p>
<p><span class="math display">\[khv = [(veg_{prim} \times 3) +
(veg_{sec} \times 2) + veg_{supp}] + (soil \times 2) +\]</span> <span
class="math display">\[+ (veg_{water} \times 2) + koala + (urb \times 3)
+ (veg_{patch} \times 3)\]</span> where:</p>
<ul>
<li><span class="math inline">\(veg_{prim}\)</span> : primary feed
trees</li>
<li><span class="math inline">\(veg_{sec}\)</span> : secondary feed
trees</li>
<li><span class="math inline">\(veg_{supp}\)</span> : supplementary
trees</li>
<li><span class="math inline">\(soil\)</span> : soil fertility</li>
<li><span class="math inline">\(veg_{water}\)</span> : vegetation
proximity to water</li>
<li><span class="math inline">\(koala\)</span> : buffered koala
sightings</li>
<li><span class="math inline">\(urb\)</span> : distance to
infrastructure</li>
<li><span class="math inline">\(veg_{patch}\)</span> : vegetation patch
size</li>
</ul>
<p>The final score is then normalised to provide a habitat value index
from 0 to 100:</p>
<ul>
<li><p><strong>Lower koala habitat value (0 – 19)</strong> – Areas that
have little or no identified mapped values for koala habitat within the
landscape. The majority of these areas are highly disturbed, fragmented
or urbanised.</p></li>
<li><p><strong>Moderate koala habitat value (20 – 33)</strong> – Areas
that have some mapped koala habitat values within the landscape. In most
cases the values within this category will provide supporting habitat
for koala in the area.</p></li>
<li><p><strong>High koala habitat value (34 – 48)</strong> – Areas with
priority koala habitat values. These areas will include a large
proportion of the criteria for koala habitat and provide an important
resource for koalas in the area.</p></li>
<li><p><strong>Very High koala habitat value (49 – 100)</strong> – An
accumulation of priority values for koala habitat. These areas contain
the majority of or even all values identified as criteria meeting
priority koala habitat within the Lower Hunter area.</p></li>
</ul>
</div>
</div>
</div>
<div id="derive-missing-data-layers" class="section level1">
<h1>Derive missing data layers</h1>
<div id="load-necessary-r-packages" class="section level2">
<h2>Load necessary R packages</h2>
<p>All necessary packages should already be installed. However, if any
are missing, you can install them using the
<code>install.packages()</code> function in R:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="co"># Main spatial packages</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;sf&quot;</span>)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;terra&quot;</span>)</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="co"># tmap dev version</span></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;tmap&quot;</span>, <span class="at">repos =</span> <span class="fu">c</span>(<span class="st">&quot;https://r-tmap.r-universe.dev&quot;</span>,</span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a>                                   <span class="st">&quot;https://cloud.r-project.org&quot;</span>))</span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a><span class="co"># leaflet for dynamic maps</span></span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;leaflet&quot;</span></span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a><span class="co"># tidyverse for additional data wrangling</span></span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;tidyverse&quot;</span>)</span></code></pre></div>
<p>Load packages:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">library</span>(sf)</span></code></pre></div>
<pre><code>## Linking to GEOS 3.11.0, GDAL 3.5.3, PROJ 9.1.0; sf_use_s2() is TRUE</code></pre>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="fu">library</span>(terra)</span></code></pre></div>
<pre><code>## terra 1.8.15</code></pre>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="fu">library</span>(tmap)</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a><span class="fu">library</span>(leaflet)</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code></pre></div>
<pre><code>## ── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
## ✔ dplyr     1.1.4     ✔ readr     2.1.5
## ✔ forcats   1.0.0     ✔ stringr   1.5.1
## ✔ ggplot2   3.5.1     ✔ tibble    3.2.1
## ✔ lubridate 1.9.4     ✔ tidyr     1.3.1
## ✔ purrr     1.0.2</code></pre>
<pre><code>## ── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
## ✖ tidyr::extract() masks terra::extract()
## ✖ dplyr::filter()  masks stats::filter()
## ✖ dplyr::lag()     masks stats::lag()
## ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="co"># Set tmap to static maps</span></span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a><span class="fu">tmap_mode</span>(<span class="st">&quot;plot&quot;</span>)</span></code></pre></div>
<pre><code>## ℹ tmap mode set to &quot;plot&quot;.</code></pre>
</div>
<div id="data-files-provided" class="section level2">
<h2>Data files provided</h2>
<p>The following data files are provided:</p>
<p><strong>Vector data</strong>:</p>
<ul>
<li><p><code>built_areas</code>: a polygon ESRI shapefile representing
various ‘built’ land uses in the study area. Data was obtained from <a
href="https://www.openstreetmap.org">OpenStreetMap</a>.</p></li>
<li><p><code>koala_sightings</code>: a point ESRI shapefile representing
koala sightings in the study area. Data was obtained from the <a
href="https://www2.environment.nsw.gov.au/topics/animals-and-plants/biodiversity/nsw-bionet">BioNet
Atlas of NSW Wildlife</a> hosted by the <em>NSW Office of Environment
and Heritage</em>.</p></li>
<li><p><code>railways</code> : a line ESRI shapefile representing
railway tracks in the study area. Data was obtained from <a
href="https://www.openstreetmap.org">OpenStreetMap</a>.</p></li>
<li><p><code>roads</code> : a line ESRI shapefile representing roads in
the study area. Data was obtained from the <em>Geoscience Australia</em>
<a
href="https://data.gov.au/data/dataset/geodata-topo-2-5m-2003-ga">GEODATA
TOPO 2.5M 2003 data product</a>.</p></li>
<li><p><code>soil_landscapes</code> : a polygon ESRI shapefile
representing soil landscapes. The data is based on digital maps from the
<em>NSW Office of Environment and Heritage</em>. The following map
sheets were combined together: <em>Soil Landscapes of the Gosford-Lake
Macquarie</em> (1:100,000), <em>Soil Landscapes of the Newcastle</em>
(1:100,000), and <em>Soil Landscapes of the Port Stephens</em>
(1:100,000).</p></li>
<li><p><code>study_area_bnd</code> : a polygon ESRI shapefile
representing the the study area, obtained by merging the administrative
boundaries of the three LGA’s considered here: Lake Macquarie,
Newcastle, and Port Stephens.</p></li>
</ul>
<p><strong>Raster data</strong>:</p>
<ul>
<li><p><code>allveg_patch_size</code>: a geoTIFF raster representing
size of vegetation patch. Data obtained from <em>Eco
Logical</em>.</p></li>
<li><p><code>allveg_prim</code>, <code>allveg_sec</code>, and
<code>allveg_supp</code>: geoTIFF rasters representing koala feed trees.
Data obtained from <em>Eco Logical</em>.</p></li>
<li><p><code>dist_to_waters</code>: a geoTIFF raster representing
vegetation proximity to water. Data obtained from <em>Eco
Logical</em>.</p></li>
</ul>
</div>
<div id="layer-1-soil-fertility" class="section level2">
<h2>Layer 1: Soil fertility</h2>
<p>Load the soil landscapes dataset (<code>soil_landscapes</code>) along
with the polygon shapefile representing the study area boundary
(<code>study_area_bnd</code>):</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="co"># Read in shapefiles</span></span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>soil_landscapes <span class="ot">&lt;-</span> <span class="fu">read_sf</span>(<span class="st">&quot;./data/data_koala/vector/soil_landscapes.shp&quot;</span>)</span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a>hunter_bnd <span class="ot">&lt;-</span> <span class="fu">read_sf</span>(<span class="st">&quot;./data/data_koala/vector/study_area_bnd.shp&quot;</span>)</span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a></span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a><span class="co"># Print the first rows of the soil landscapes layer</span></span>
<span id="cb11-6"><a href="#cb11-6" tabindex="-1"></a><span class="fu">head</span>(soil_landscapes)</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["LANDSCAPE"],"name":[1],"type":["chr"],"align":["left"]},{"label":["NAME"],"name":[2],"type":["chr"],"align":["left"]},{"label":["CODE"],"name":[3],"type":["chr"],"align":["left"]},{"label":["PROCESS"],"name":[4],"type":["chr"],"align":["left"]},{"label":["geometry"],"name":[5],"type":["s_MULTIP"],"align":["right"]}],"data":[{"1":"AEsba","2":"SHOAL BAY variant a","3":"sba","4":"AEOLIAN","5":"<s_MULTIP>"},{"1":"ESfc","2":"FULLERTON COVE","3":"fc","4":"ESTUARINE","5":"<s_MULTIP>"},{"1":"ESfc","2":"FULLERTON COVE","3":"fc","4":"ESTUARINE","5":"<s_MULTIP>"},{"1":"AEbt","2":"BOYCES TRACK","3":"bt","4":"AEOLIAN","5":"<s_MULTIP>"},{"1":"TRng","2":"NUNGRA","3":"ng","4":"TRANSFERRAL","5":"<s_MULTIP>"},{"1":"ESfc","2":"FULLERTON COVE","3":"fc","4":"ESTUARINE","5":"<s_MULTIP>"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>The soil landscapes layer includes four attribute columns, the forth
(<code>PROCESS</code>) representing the geomorphic type of landscape on
which the soils have formed.</p>
<p>Plot the <code>soil_landscapes</code> layer, colouring each polygon
according to the value of the <code>PROCESS</code> column:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="co"># Plot data for quick visual inspection</span></span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a>soils_map <span class="ot">&lt;-</span> <span class="fu">tm_shape</span>(soil_landscapes) <span class="sc">+</span> </span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a>        <span class="fu">tm_fill</span>(<span class="st">&quot;PROCESS&quot;</span>) <span class="sc">+</span> </span>
<span id="cb12-4"><a href="#cb12-4" tabindex="-1"></a>        <span class="fu">tm_layout</span>(<span class="at">frame =</span> <span class="cn">FALSE</span>, <span class="at">legend.text.size =</span> <span class="fl">0.5</span>, <span class="at">legend.title.size =</span> <span class="fl">0.5</span>)</span>
<span id="cb12-5"><a href="#cb12-5" tabindex="-1"></a></span>
<span id="cb12-6"><a href="#cb12-6" tabindex="-1"></a><span class="co"># Plot study area bnd</span></span>
<span id="cb12-7"><a href="#cb12-7" tabindex="-1"></a>soils_map <span class="sc">+</span> <span class="fu">tm_shape</span>(hunter_bnd) <span class="sc">+</span>  <span class="fu">tm_borders</span>(<span class="st">&quot;black&quot;</span>) </span></code></pre></div>
<p><img src="r5_files/figure-html/unnamed-chunk-13-1.png" width="100%" style="display: block; margin: auto;" /></p>
<p>To clean up the data, we will first crop <code>soil_landscapes</code>
to the extent of the study area (i.e., <code>hunter_bnd</code>). Next,
we will isolate those soil landscapes that can sustain koala feed tree
species. These soils include those formed on Quaternary alluvial,
colluvial, and aeolian deposits, and those formed in swampy areas.</p>
<p>To crop the <code>soil_landscapes</code> layer will use the
<strong>sf</strong> function <code>st_intersection()</code>:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a><span class="co"># Intersect the two polygon layers</span></span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a>soil_landscapes_bnd <span class="ot">&lt;-</span> <span class="fu">st_intersection</span>(soil_landscapes, hunter_bnd)</span></code></pre></div>
<pre><code>## Warning: attribute variables are assumed to be spatially constant throughout
## all geometries</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a><span class="co"># Plot data for quick visual inspection</span></span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a><span class="fu">tm_shape</span>(soil_landscapes_bnd) <span class="sc">+</span> </span>
<span id="cb15-3"><a href="#cb15-3" tabindex="-1"></a>  <span class="fu">tm_fill</span>(<span class="st">&quot;PROCESS&quot;</span>) <span class="sc">+</span> </span>
<span id="cb15-4"><a href="#cb15-4" tabindex="-1"></a>  <span class="fu">tm_layout</span>(<span class="at">frame =</span> <span class="cn">FALSE</span>, <span class="at">legend.text.size =</span> <span class="fl">0.6</span>, <span class="at">legend.title.size =</span> <span class="fl">0.6</span>)</span></code></pre></div>
<p><img src="r5_files/figure-html/unnamed-chunk-14-1.png" width="90%" style="display: block; margin: auto;" /></p>
<p>Next we extract the soil landscapes of interest. We use the
<strong>dplyr</strong> function <code>filter()</code>:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a><span class="co"># Create list of PROCESS values of interest</span></span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a>soils <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;AEOLIAN&quot;</span>, <span class="st">&quot;ALLUVIAL&quot;</span>, <span class="st">&quot;COLLUVIAL&quot;</span>, <span class="st">&quot;SWAMP&quot;</span>, <span class="st">&quot;RESIDUAL&quot;</span>, <span class="st">&quot;ESTUARINE&quot;</span>)</span>
<span id="cb16-3"><a href="#cb16-3" tabindex="-1"></a></span>
<span id="cb16-4"><a href="#cb16-4" tabindex="-1"></a><span class="co"># Filter data and extract polygons of interest</span></span>
<span id="cb16-5"><a href="#cb16-5" tabindex="-1"></a>soil_landscapes_filter <span class="ot">&lt;-</span> <span class="fu">filter</span>(soil_landscapes_bnd, PROCESS <span class="sc">%in%</span> soils)</span>
<span id="cb16-6"><a href="#cb16-6" tabindex="-1"></a></span>
<span id="cb16-7"><a href="#cb16-7" tabindex="-1"></a><span class="co"># Plot study area bnd first as it has larger extent</span></span>
<span id="cb16-8"><a href="#cb16-8" tabindex="-1"></a>soils_map2 <span class="ot">&lt;-</span> <span class="fu">tm_shape</span>(hunter_bnd) <span class="sc">+</span>  <span class="fu">tm_borders</span>(<span class="st">&quot;gray&quot;</span>)</span>
<span id="cb16-9"><a href="#cb16-9" tabindex="-1"></a></span>
<span id="cb16-10"><a href="#cb16-10" tabindex="-1"></a><span class="co"># Add soils  </span></span>
<span id="cb16-11"><a href="#cb16-11" tabindex="-1"></a>soils_map2 <span class="sc">+</span> <span class="fu">tm_shape</span>(soil_landscapes_filter) <span class="sc">+</span> </span>
<span id="cb16-12"><a href="#cb16-12" tabindex="-1"></a>      <span class="fu">tm_fill</span>(<span class="st">&quot;PROCESS&quot;</span>) <span class="sc">+</span> </span>
<span id="cb16-13"><a href="#cb16-13" tabindex="-1"></a>      <span class="fu">tm_layout</span>(<span class="at">frame =</span> <span class="cn">FALSE</span>, <span class="at">legend.text.size =</span> <span class="fl">0.6</span>, <span class="at">legend.title.size =</span> <span class="fl">0.6</span>)</span></code></pre></div>
<p><img src="r5_files/figure-html/unnamed-chunk-15-1.png" width="80%" style="display: block; margin: auto;" /></p>
<p>The next chunk of code adds a new attribute column (called
<code>SCORE</code>) and uses the <strong>dplyr</strong> function
<code>mutate()</code> to assign values based on another column (in this
case <code>PROCESS</code>). We assign the value 100 to each polygon
where <code>PROCESS</code> equals “AEOLIAN”, “ALLUVIAL”, “COLLUVIAL”,
“SWAMP”, “RESIDUAL”, or “ESTUARINE”.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a><span class="co"># Add a new numeric field &#39;SCORE&#39; based on conditions applied to &#39;PROCESS&#39;</span></span>
<span id="cb17-2"><a href="#cb17-2" tabindex="-1"></a><span class="co"># Here, we assign:</span></span>
<span id="cb17-3"><a href="#cb17-3" tabindex="-1"></a><span class="co">#   100 if &#39;PROCESS&#39; equals &quot;AEOLIAN&quot;, &quot;ALLUVIAL&quot;, &quot;COLLUVIAL&quot;, ...</span></span>
<span id="cb17-4"><a href="#cb17-4" tabindex="-1"></a><span class="co">#   0 for all other cases</span></span>
<span id="cb17-5"><a href="#cb17-5" tabindex="-1"></a>soil_landscapes_filter <span class="ot">&lt;-</span> soil_landscapes_filter <span class="sc">%&gt;%</span></span>
<span id="cb17-6"><a href="#cb17-6" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SCORE =</span> <span class="fu">case_when</span>(</span>
<span id="cb17-7"><a href="#cb17-7" tabindex="-1"></a>    PROCESS <span class="sc">==</span> <span class="st">&quot;AEOLIAN&quot;</span> <span class="sc">~</span> <span class="dv">100</span>,</span>
<span id="cb17-8"><a href="#cb17-8" tabindex="-1"></a>    PROCESS <span class="sc">==</span> <span class="st">&quot;ALLUVIAL&quot;</span> <span class="sc">~</span> <span class="dv">100</span>,</span>
<span id="cb17-9"><a href="#cb17-9" tabindex="-1"></a>    PROCESS <span class="sc">==</span> <span class="st">&quot;COLLUVIAL&quot;</span> <span class="sc">~</span> <span class="dv">100</span>,</span>
<span id="cb17-10"><a href="#cb17-10" tabindex="-1"></a>    PROCESS <span class="sc">==</span> <span class="st">&quot;SWAMP&quot;</span> <span class="sc">~</span> <span class="dv">100</span>,</span>
<span id="cb17-11"><a href="#cb17-11" tabindex="-1"></a>    PROCESS <span class="sc">==</span> <span class="st">&quot;RESIDUAL&quot;</span> <span class="sc">~</span> <span class="dv">100</span>,</span>
<span id="cb17-12"><a href="#cb17-12" tabindex="-1"></a>    PROCESS <span class="sc">==</span> <span class="st">&quot;ESTUARINE&quot;</span> <span class="sc">~</span> <span class="dv">100</span>,</span>
<span id="cb17-13"><a href="#cb17-13" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> <span class="dv">0</span>  <span class="co"># default numeric value for unmatched cases</span></span>
<span id="cb17-14"><a href="#cb17-14" tabindex="-1"></a>  ))</span>
<span id="cb17-15"><a href="#cb17-15" tabindex="-1"></a></span>
<span id="cb17-16"><a href="#cb17-16" tabindex="-1"></a><span class="co"># Plot study area bnd first as it has larger extent</span></span>
<span id="cb17-17"><a href="#cb17-17" tabindex="-1"></a>soils_map3 <span class="ot">&lt;-</span> <span class="fu">tm_shape</span>(hunter_bnd) <span class="sc">+</span>  <span class="fu">tm_borders</span>(<span class="st">&quot;gray&quot;</span>)</span>
<span id="cb17-18"><a href="#cb17-18" tabindex="-1"></a></span>
<span id="cb17-19"><a href="#cb17-19" tabindex="-1"></a><span class="co"># Add soils  </span></span>
<span id="cb17-20"><a href="#cb17-20" tabindex="-1"></a>soils_map3 <span class="sc">+</span>  <span class="fu">tm_shape</span>(soil_landscapes_filter) <span class="sc">+</span> </span>
<span id="cb17-21"><a href="#cb17-21" tabindex="-1"></a>        <span class="fu">tm_fill</span>(<span class="st">&quot;SCORE&quot;</span>) <span class="sc">+</span> </span>
<span id="cb17-22"><a href="#cb17-22" tabindex="-1"></a>        <span class="fu">tm_layout</span>(<span class="at">frame =</span> <span class="cn">FALSE</span>, <span class="at">legend.text.size =</span> <span class="fl">0.6</span>, <span class="at">legend.title.size =</span> <span class="fl">0.6</span>)</span></code></pre></div>
<pre><code>## The visual variable &quot;fill&quot; of the layer &quot;polygons&quot; contains a unique value. Therefore a discrete scale is applied (tm_scale_discrete).</code></pre>
<p><img src="r5_files/figure-html/unnamed-chunk-16-1.png" width="80%" style="display: block; margin: auto;" /></p>
<p>The last step will convert <code>soil_landscapes_filter</code> to a
raster. This will be our final soil fertility layer:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a><span class="co"># Convert the sf object to a terra SpatVector</span></span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a>soil_landscapes_terra <span class="ot">&lt;-</span> <span class="fu">vect</span>(soil_landscapes_filter)</span>
<span id="cb19-3"><a href="#cb19-3" tabindex="-1"></a></span>
<span id="cb19-4"><a href="#cb19-4" tabindex="-1"></a><span class="co"># Read in the target SpatRaster that defines the extent and resolution</span></span>
<span id="cb19-5"><a href="#cb19-5" tabindex="-1"></a><span class="co"># Here we read in one of the Eco Logical geoTIFFs provided</span></span>
<span id="cb19-6"><a href="#cb19-6" tabindex="-1"></a>target_raster <span class="ot">&lt;-</span> <span class="fu">rast</span>(<span class="st">&quot;./data/data_koala/raster/allveg_patch_size.tif&quot;</span>)</span>
<span id="cb19-7"><a href="#cb19-7" tabindex="-1"></a></span>
<span id="cb19-8"><a href="#cb19-8" tabindex="-1"></a><span class="co"># Rasterise the polygons using an attribute field</span></span>
<span id="cb19-9"><a href="#cb19-9" tabindex="-1"></a>soil_landscapes_rast <span class="ot">&lt;-</span> <span class="fu">rasterize</span>(soil_landscapes_terra, target_raster, <span class="at">field =</span> <span class="st">&quot;SCORE&quot;</span>)</span>
<span id="cb19-10"><a href="#cb19-10" tabindex="-1"></a></span>
<span id="cb19-11"><a href="#cb19-11" tabindex="-1"></a><span class="co"># Convert all NA (no data) values in the new raster to 0</span></span>
<span id="cb19-12"><a href="#cb19-12" tabindex="-1"></a>soil_landscapes_rast[<span class="fu">is.na</span>(soil_landscapes_rast)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb19-13"><a href="#cb19-13" tabindex="-1"></a></span>
<span id="cb19-14"><a href="#cb19-14" tabindex="-1"></a><span class="co"># Plot data for quick visual inspection</span></span>
<span id="cb19-15"><a href="#cb19-15" tabindex="-1"></a>soils_map4 <span class="ot">&lt;-</span> <span class="fu">tm_shape</span>(soil_landscapes_rast) <span class="sc">+</span> </span>
<span id="cb19-16"><a href="#cb19-16" tabindex="-1"></a>       <span class="fu">tm_raster</span>(<span class="at">col.scale =</span> <span class="fu">tm_scale_categorical</span>()) <span class="sc">+</span> </span>
<span id="cb19-17"><a href="#cb19-17" tabindex="-1"></a>       <span class="fu">tm_layout</span>(<span class="at">frame =</span> <span class="cn">FALSE</span>, <span class="at">legend.text.size =</span> <span class="fl">0.6</span>, <span class="at">legend.title.size =</span> <span class="fl">0.6</span>)</span>
<span id="cb19-18"><a href="#cb19-18" tabindex="-1"></a></span>
<span id="cb19-19"><a href="#cb19-19" tabindex="-1"></a><span class="co"># Plot study area bnd on top</span></span>
<span id="cb19-20"><a href="#cb19-20" tabindex="-1"></a>soils_map4 <span class="sc">+</span> <span class="fu">tm_shape</span>(hunter_bnd) <span class="sc">+</span>  <span class="fu">tm_borders</span>(<span class="st">&quot;white&quot;</span>) </span></code></pre></div>
<pre><code>## SpatRaster object downsampled to 3018 by 3314 cells.</code></pre>
<p><img src="r5_files/figure-html/unnamed-chunk-17-1.png" width="90%" style="display: block; margin: auto;" /></p>
</div>
<div id="layer-2-koala-sightings" class="section level2">
<h2>Layer 2: Koala sightings</h2>
<p>Load the koala sightings dataset (<code>koala_sightings</code>):</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a><span class="co"># Read in shapefile</span></span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a>koala_sightings <span class="ot">&lt;-</span> <span class="fu">read_sf</span>(<span class="st">&quot;./data/data_koala/vector/koala_sightings.shp&quot;</span>)</span>
<span id="cb21-3"><a href="#cb21-3" tabindex="-1"></a></span>
<span id="cb21-4"><a href="#cb21-4" tabindex="-1"></a><span class="co"># Print the first rows of the soil landscapes layer</span></span>
<span id="cb21-5"><a href="#cb21-5" tabindex="-1"></a><span class="fu">head</span>(koala_sightings)</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["SightingKe"],"name":[1],"type":["chr"],"align":["left"]},{"label":["SpeciesCod"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["KingdomNam"],"name":[3],"type":["chr"],"align":["left"]},{"label":["ClassName"],"name":[4],"type":["chr"],"align":["left"]},{"label":["FamilyName"],"name":[5],"type":["chr"],"align":["left"]},{"label":["SortOrder"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["Scientific"],"name":[7],"type":["chr"],"align":["left"]},{"label":["CommonName"],"name":[8],"type":["chr"],"align":["left"]},{"label":["ProfileID"],"name":[9],"type":["dbl"],"align":["right"]},{"label":["DateFirst"],"name":[10],"type":["date"],"align":["right"]},{"label":["DateLast"],"name":[11],"type":["date"],"align":["right"]},{"label":["NumberIndi"],"name":[12],"type":["dbl"],"align":["right"]},{"label":["Observatio"],"name":[13],"type":["chr"],"align":["left"]},{"label":["Latitude_G"],"name":[14],"type":["dbl"],"align":["right"]},{"label":["Longitude_"],"name":[15],"type":["dbl"],"align":["right"]},{"label":["Zone"],"name":[16],"type":["dbl"],"align":["right"]},{"label":["Easting"],"name":[17],"type":["dbl"],"align":["right"]},{"label":["Northing"],"name":[18],"type":["dbl"],"align":["right"]},{"label":["Accuracy"],"name":[19],"type":["dbl"],"align":["right"]},{"label":["DateRank"],"name":[20],"type":["int"],"align":["right"]},{"label":["geometry"],"name":[21],"type":["sf_POINT"],"align":["right"]}],"data":[{"1":"WS-02847","2":"1162","3":"Fauna","4":"Mammalia","5":"Phascolarctidae","6":"3434","7":"Phascolarctos cinereus","8":"Koala","9":"10616","10":"2004-07-01","11":"2006-06-30","12":"0","13":"O","14":"-32.72960","15":"151.9979","16":"56","17":"406100","18":"6378245","19":"10000","20":"100","21":"<sf_POINT>"},{"1":"WS-02849","2":"1162","3":"Fauna","4":"Mammalia","5":"Phascolarctidae","6":"3434","7":"Phascolarctos cinereus","8":"Koala","9":"10616","10":"1980-01-01","11":"2004-06-30","12":"1","13":"R","14":"-32.77903","15":"151.8794","16":"56","17":"395054","18":"6372654","19":"10000","20":"100","21":"<sf_POINT>"},{"1":"WS-02850","2":"1162","3":"Fauna","4":"Mammalia","5":"Phascolarctidae","6":"3434","7":"Phascolarctos cinereus","8":"Koala","9":"10616","10":"1980-01-01","11":"2004-06-30","12":"2","13":"R","14":"-32.78438","15":"151.8954","16":"56","17":"396564","18":"6372076","19":"10000","20":"100","21":"<sf_POINT>"},{"1":"WS-02851","2":"1162","3":"Fauna","4":"Mammalia","5":"Phascolarctidae","6":"3434","7":"Phascolarctos cinereus","8":"Koala","9":"10616","10":"1980-01-01","11":"2004-06-30","12":"3","13":"R","14":"-32.79152","15":"151.9115","16":"56","17":"398077","18":"6371300","19":"10000","20":"100","21":"<sf_POINT>"},{"1":"WS-02852","2":"1162","3":"Fauna","4":"Mammalia","5":"Phascolarctidae","6":"3434","7":"Phascolarctos cinereus","8":"Koala","9":"10616","10":"1980-01-01","11":"2004-06-30","12":"4","13":"R","14":"-32.74512","15":"151.9597","16":"56","17":"402538","18":"6376490","19":"10000","20":"100","21":"<sf_POINT>"},{"1":"WS-02853","2":"1162","3":"Fauna","4":"Mammalia","5":"Phascolarctidae","6":"3434","7":"Phascolarctos cinereus","8":"Koala","9":"10616","10":"1980-01-01","11":"2004-06-30","12":"5","13":"R","14":"-32.75583","15":"151.9734","16":"56","17":"403832","18":"6375315","19":"10000","20":"100","21":"<sf_POINT>"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>The koala sightings layer contains 20 attribute fields, with the last
one (<code>DateRank</code>) containing two values: ‘75’ for all
sightings made before 1985, and ‘100’ for all koala sightings dated 1986
and later. Data on sightings reported before 1985 are considered less
reliable than those made after, explaining the lower rank.</p>
<p>Plot the <code>koala_sightings</code> layer, colouring each point
according to the value of the <code>DateRank</code> column:</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a><span class="co"># Plot study area bnd first as it has larger extent</span></span>
<span id="cb22-2"><a href="#cb22-2" tabindex="-1"></a>koala_map <span class="ot">&lt;-</span> <span class="fu">tm_shape</span>(hunter_bnd) <span class="sc">+</span>  <span class="fu">tm_borders</span>(<span class="st">&quot;black&quot;</span>)</span>
<span id="cb22-3"><a href="#cb22-3" tabindex="-1"></a></span>
<span id="cb22-4"><a href="#cb22-4" tabindex="-1"></a><span class="co"># Add koalas</span></span>
<span id="cb22-5"><a href="#cb22-5" tabindex="-1"></a>koala_map <span class="sc">+</span> <span class="fu">tm_shape</span>(koala_sightings) <span class="sc">+</span> </span>
<span id="cb22-6"><a href="#cb22-6" tabindex="-1"></a>        <span class="fu">tm_dots</span>(<span class="at">size =</span> <span class="fl">0.2</span>, <span class="at">fill =</span> <span class="st">&quot;DateRank&quot;</span>, <span class="at">fill.scale =</span> <span class="fu">tm_scale_categorical</span>()) <span class="sc">+</span> </span>
<span id="cb22-7"><a href="#cb22-7" tabindex="-1"></a>        <span class="fu">tm_layout</span>(<span class="at">frame =</span> <span class="cn">FALSE</span>, <span class="at">legend.text.size =</span> <span class="fl">0.5</span>, <span class="at">legend.title.size =</span> <span class="fl">0.5</span>)</span></code></pre></div>
<p><img src="r5_files/figure-html/unnamed-chunk-19-1.png" width="80%" style="display: block; margin: auto;" /></p>
<p>In the next step, we drop all columns except <code>DateRank</code>
and create a 100 m buffer around each point using
<code>st_buffer()</code>. Next, we dissolve the layer using
<code>st_union()</code> so that all features are grouped together based
on their value for <code>DateRank</code>. Lastly, we rename
<code>DateRank</code> to <code>SCORE</code>.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a><span class="co"># Retain only the &quot;DateRank&quot; attribute column (geometry is preserved automatically)</span></span>
<span id="cb23-2"><a href="#cb23-2" tabindex="-1"></a>koala_sightings <span class="ot">&lt;-</span> koala_sightings <span class="sc">%&gt;%</span> </span>
<span id="cb23-3"><a href="#cb23-3" tabindex="-1"></a>  <span class="fu">select</span>(DateRank)</span>
<span id="cb23-4"><a href="#cb23-4" tabindex="-1"></a></span>
<span id="cb23-5"><a href="#cb23-5" tabindex="-1"></a><span class="co"># Create a 100m buffer around each point</span></span>
<span id="cb23-6"><a href="#cb23-6" tabindex="-1"></a>koala_buffers <span class="ot">&lt;-</span> <span class="fu">st_buffer</span>(koala_sightings, <span class="at">dist =</span> <span class="dv">100</span>)</span>
<span id="cb23-7"><a href="#cb23-7" tabindex="-1"></a></span>
<span id="cb23-8"><a href="#cb23-8" tabindex="-1"></a><span class="co"># Dissolve (union) buffers by the DateRank field and rename field to SCORE</span></span>
<span id="cb23-9"><a href="#cb23-9" tabindex="-1"></a>koala_buffers_dissolved <span class="ot">&lt;-</span> koala_buffers <span class="sc">%&gt;%</span></span>
<span id="cb23-10"><a href="#cb23-10" tabindex="-1"></a>  <span class="fu">group_by</span>(DateRank) <span class="sc">%&gt;%</span></span>
<span id="cb23-11"><a href="#cb23-11" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">geometry =</span> <span class="fu">st_union</span>(geometry), <span class="at">.groups =</span> <span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb23-12"><a href="#cb23-12" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">SCORE =</span> DateRank)</span>
<span id="cb23-13"><a href="#cb23-13" tabindex="-1"></a></span>
<span id="cb23-14"><a href="#cb23-14" tabindex="-1"></a><span class="co"># Plot study area bnd first as it has larger extent</span></span>
<span id="cb23-15"><a href="#cb23-15" tabindex="-1"></a>koala_map2 <span class="ot">&lt;-</span> <span class="fu">tm_shape</span>(hunter_bnd) <span class="sc">+</span>  <span class="fu">tm_borders</span>(<span class="st">&quot;black&quot;</span>)</span>
<span id="cb23-16"><a href="#cb23-16" tabindex="-1"></a></span>
<span id="cb23-17"><a href="#cb23-17" tabindex="-1"></a><span class="co"># Add koalas</span></span>
<span id="cb23-18"><a href="#cb23-18" tabindex="-1"></a>koala_map2 <span class="sc">+</span> <span class="fu">tm_shape</span>(koala_buffers_dissolved) <span class="sc">+</span> </span>
<span id="cb23-19"><a href="#cb23-19" tabindex="-1"></a>        <span class="fu">tm_fill</span>(<span class="at">fill =</span> <span class="st">&quot;SCORE&quot;</span>, <span class="at">fill.scale =</span> <span class="fu">tm_scale_categorical</span>()) <span class="sc">+</span> </span>
<span id="cb23-20"><a href="#cb23-20" tabindex="-1"></a>        <span class="fu">tm_layout</span>(<span class="at">frame =</span> <span class="cn">FALSE</span>, <span class="at">legend.text.size =</span> <span class="fl">0.5</span>, <span class="at">legend.title.size =</span> <span class="fl">0.5</span>)</span></code></pre></div>
<p><img src="r5_files/figure-html/unnamed-chunk-20-1.png" width="80%" style="display: block; margin: auto;" /></p>
<p>We are now ready to convert the koala buffer layer
(<code>koala_buffers_dissolved</code>) to a raster with scores:</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a><span class="co"># Convert the sf object to a terra SpatVector</span></span>
<span id="cb24-2"><a href="#cb24-2" tabindex="-1"></a>koala_buffers_terra <span class="ot">&lt;-</span> <span class="fu">vect</span>(koala_buffers_dissolved)</span>
<span id="cb24-3"><a href="#cb24-3" tabindex="-1"></a></span>
<span id="cb24-4"><a href="#cb24-4" tabindex="-1"></a><span class="co"># Read in the target SpatRaster that defines the extent and resolution</span></span>
<span id="cb24-5"><a href="#cb24-5" tabindex="-1"></a><span class="co"># Here we read in one of the Eco Logical geoTIFFs provided</span></span>
<span id="cb24-6"><a href="#cb24-6" tabindex="-1"></a>target_raster <span class="ot">&lt;-</span> <span class="fu">rast</span>(<span class="st">&quot;./data/data_koala/raster/allveg_patch_size.tif&quot;</span>)</span>
<span id="cb24-7"><a href="#cb24-7" tabindex="-1"></a></span>
<span id="cb24-8"><a href="#cb24-8" tabindex="-1"></a><span class="co"># Rasterise the polygons using an attribute field</span></span>
<span id="cb24-9"><a href="#cb24-9" tabindex="-1"></a>koala_buffers_rast <span class="ot">&lt;-</span> <span class="fu">rasterize</span>(koala_buffers_terra, target_raster, <span class="at">field =</span> <span class="st">&quot;SCORE&quot;</span>)</span>
<span id="cb24-10"><a href="#cb24-10" tabindex="-1"></a></span>
<span id="cb24-11"><a href="#cb24-11" tabindex="-1"></a><span class="co"># Convert all NA (no data) values in the new raster to 0</span></span>
<span id="cb24-12"><a href="#cb24-12" tabindex="-1"></a>koala_buffers_rast[<span class="fu">is.na</span>(koala_buffers_rast)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb24-13"><a href="#cb24-13" tabindex="-1"></a></span>
<span id="cb24-14"><a href="#cb24-14" tabindex="-1"></a><span class="co"># Plot data for quick visual inspection</span></span>
<span id="cb24-15"><a href="#cb24-15" tabindex="-1"></a>koala_map3 <span class="ot">&lt;-</span> <span class="fu">tm_shape</span>(koala_buffers_rast) <span class="sc">+</span> </span>
<span id="cb24-16"><a href="#cb24-16" tabindex="-1"></a>       <span class="fu">tm_raster</span>(<span class="at">col.scale =</span> <span class="fu">tm_scale_categorical</span>()) <span class="sc">+</span> </span>
<span id="cb24-17"><a href="#cb24-17" tabindex="-1"></a>       <span class="fu">tm_layout</span>(<span class="at">frame =</span> <span class="cn">FALSE</span>, <span class="at">legend.text.size =</span> <span class="fl">0.6</span>, <span class="at">legend.title.size =</span> <span class="fl">0.6</span>)</span>
<span id="cb24-18"><a href="#cb24-18" tabindex="-1"></a></span>
<span id="cb24-19"><a href="#cb24-19" tabindex="-1"></a><span class="co"># Plot study area bnd on top</span></span>
<span id="cb24-20"><a href="#cb24-20" tabindex="-1"></a>koala_map3 <span class="sc">+</span> <span class="fu">tm_shape</span>(hunter_bnd) <span class="sc">+</span>  <span class="fu">tm_borders</span>(<span class="st">&quot;white&quot;</span>) </span></code></pre></div>
<pre><code>## SpatRaster object downsampled to 3018 by 3314 cells.</code></pre>
<p><img src="r5_files/figure-html/unnamed-chunk-21-1.png" width="80%" style="display: block; margin: auto;" /></p>
</div>
<div id="layer-3-distance-to-infrastructure" class="section level2">
<h2>Layer 3: Distance to infrastructure</h2>
<p>For our final layer, we need to calculate a 300 m buffer around
infrastructure that may represent a barrier for koala movement, and
assign a score of 100 to all areas that are beyond this 300 m buffer.
Here, we will consider the following types of infrastructure as barriers
to koala movement: roads, railway corridors, and built-up areas,
including those classified as developed land even if they lack actual
buildings.</p>
<p>Load the relevant datasets, namely, <code>built_areas</code>,
<code>roads</code>, and <code>railways</code>:</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a><span class="co"># Read in shapefiles</span></span>
<span id="cb26-2"><a href="#cb26-2" tabindex="-1"></a>built_areas <span class="ot">&lt;-</span> <span class="fu">read_sf</span>(<span class="st">&quot;./data/data_koala/vector/built_areas.shp&quot;</span>)</span>
<span id="cb26-3"><a href="#cb26-3" tabindex="-1"></a>roads <span class="ot">&lt;-</span> <span class="fu">read_sf</span>(<span class="st">&quot;./data/data_koala/vector/roads.shp&quot;</span>)</span>
<span id="cb26-4"><a href="#cb26-4" tabindex="-1"></a>railways <span class="ot">&lt;-</span> <span class="fu">read_sf</span>(<span class="st">&quot;./data/data_koala/vector/railways.shp&quot;</span>)</span>
<span id="cb26-5"><a href="#cb26-5" tabindex="-1"></a></span>
<span id="cb26-6"><a href="#cb26-6" tabindex="-1"></a><span class="co"># Plot bnd on top for larger extent</span></span>
<span id="cb26-7"><a href="#cb26-7" tabindex="-1"></a>bnd_map <span class="ot">&lt;-</span> <span class="fu">tm_shape</span>(hunter_bnd) <span class="sc">+</span>  <span class="fu">tm_borders</span>(<span class="st">&quot;black&quot;</span>) <span class="sc">+</span> <span class="fu">tm_layout</span>(<span class="at">frame =</span> <span class="cn">FALSE</span>)</span>
<span id="cb26-8"><a href="#cb26-8" tabindex="-1"></a></span>
<span id="cb26-9"><a href="#cb26-9" tabindex="-1"></a><span class="co"># Plot the three layers on the same map</span></span>
<span id="cb26-10"><a href="#cb26-10" tabindex="-1"></a></span>
<span id="cb26-11"><a href="#cb26-11" tabindex="-1"></a>built_map <span class="ot">&lt;-</span> bnd_map <span class="sc">+</span> <span class="fu">tm_shape</span>(built_areas) <span class="sc">+</span> <span class="fu">tm_fill</span>(<span class="st">&quot;lightgray&quot;</span>)</span>
<span id="cb26-12"><a href="#cb26-12" tabindex="-1"></a>roads_map <span class="ot">&lt;-</span> built_map <span class="sc">+</span> <span class="fu">tm_shape</span>(roads) <span class="sc">+</span> <span class="fu">tm_lines</span>(<span class="at">col =</span> <span class="st">&quot;red&quot;</span>, <span class="at">lwd =</span> <span class="dv">1</span>, <span class="at">lty =</span> <span class="st">&quot;solid&quot;</span>)</span>
<span id="cb26-13"><a href="#cb26-13" tabindex="-1"></a>roads_map <span class="sc">+</span> <span class="fu">tm_shape</span>(railways) <span class="sc">+</span> <span class="fu">tm_lines</span>(<span class="at">col =</span> <span class="st">&quot;black&quot;</span>, <span class="at">lwd =</span> <span class="dv">2</span>, <span class="at">lty =</span> <span class="st">&quot;solid&quot;</span>)</span></code></pre></div>
<p><img src="r5_files/figure-html/unnamed-chunk-22-1.png" width="80%" style="display: block; margin: auto;" /></p>
<p>The <code>roads</code> layer contains all road types, including minor
roads and tracks, and not all of these will constitute a barrier for
koala movement. For this reason we need to clean up this layer before
proceeding with the calculation of the buffers.</p>
<p>Print a list of the <code>roads</code> layer attribute table
columns:</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" tabindex="-1"></a><span class="fu">names</span>(roads)</span></code></pre></div>
<pre><code>##  [1] &quot;OBJECTID&quot;   &quot;FEATTYPE&quot;   &quot;NAME&quot;       &quot;CLASS&quot;      &quot;FORMATION&quot; 
##  [6] &quot;NRN&quot;        &quot;SRN&quot;        &quot;FEATREL&quot;    &quot;ATTRREL&quot;    &quot;PLANACC&quot;   
## [11] &quot;SOURCE&quot;     &quot;CREATED&quot;    &quot;RETIRED&quot;    &quot;PID&quot;        &quot;SYMBOL&quot;    
## [16] &quot;FEATWIDTH&quot;  &quot;TEXTNOTE&quot;   &quot;SHAPE_Leng&quot; &quot;geometry&quot;</code></pre>
<p>There is a column called <code>CLASS</code> and this will contain the
various road types. Print a list with the entries in this column:</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" tabindex="-1"></a><span class="fu">unique</span>(roads<span class="sc">$</span>CLASS)</span></code></pre></div>
<pre><code>## [1] &quot;Minor Road&quot;       &quot;Track&quot;            &quot;Secondary Road&quot;   &quot;Principal Road&quot;  
## [5] &quot;Dual Carriageway&quot;</code></pre>
<p>From the above, we can see that we have six road types, including
“Minor Road” and “Track”. The next chunk of code drops all roads with
these two classifications:</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" tabindex="-1"></a><span class="co"># Create list of CLASS values of interest</span></span>
<span id="cb31-2"><a href="#cb31-2" tabindex="-1"></a>road_class <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Secondary Road&quot;</span>, <span class="st">&quot;Principal Road&quot;</span>, <span class="st">&quot;Dual Carriageway&quot;</span>)</span>
<span id="cb31-3"><a href="#cb31-3" tabindex="-1"></a></span>
<span id="cb31-4"><a href="#cb31-4" tabindex="-1"></a><span class="co"># Filter data and extract roads of interest</span></span>
<span id="cb31-5"><a href="#cb31-5" tabindex="-1"></a>roads_filtered <span class="ot">&lt;-</span> <span class="fu">filter</span>(roads, CLASS <span class="sc">%in%</span> road_class)</span>
<span id="cb31-6"><a href="#cb31-6" tabindex="-1"></a></span>
<span id="cb31-7"><a href="#cb31-7" tabindex="-1"></a><span class="co"># Repeat the previous map to see roads remaining</span></span>
<span id="cb31-8"><a href="#cb31-8" tabindex="-1"></a><span class="co"># Plot bnd on top for larger extent</span></span>
<span id="cb31-9"><a href="#cb31-9" tabindex="-1"></a>bnd_map <span class="ot">&lt;-</span> <span class="fu">tm_shape</span>(hunter_bnd) <span class="sc">+</span>  <span class="fu">tm_borders</span>(<span class="st">&quot;black&quot;</span>) <span class="sc">+</span> <span class="fu">tm_layout</span>(<span class="at">frame =</span> <span class="cn">FALSE</span>)</span>
<span id="cb31-10"><a href="#cb31-10" tabindex="-1"></a></span>
<span id="cb31-11"><a href="#cb31-11" tabindex="-1"></a><span class="co"># Plot the three layers on the same map</span></span>
<span id="cb31-12"><a href="#cb31-12" tabindex="-1"></a></span>
<span id="cb31-13"><a href="#cb31-13" tabindex="-1"></a>built_map <span class="ot">&lt;-</span> bnd_map <span class="sc">+</span> <span class="fu">tm_shape</span>(built_areas) <span class="sc">+</span> <span class="fu">tm_fill</span>(<span class="st">&quot;lightgray&quot;</span>)</span>
<span id="cb31-14"><a href="#cb31-14" tabindex="-1"></a>roads_map <span class="ot">&lt;-</span> built_map <span class="sc">+</span> <span class="fu">tm_shape</span>(roads_filtered) <span class="sc">+</span> <span class="fu">tm_lines</span>(<span class="at">col =</span> <span class="st">&quot;red&quot;</span>, <span class="at">lwd =</span> <span class="dv">1</span>, <span class="at">lty =</span> <span class="st">&quot;solid&quot;</span>)</span>
<span id="cb31-15"><a href="#cb31-15" tabindex="-1"></a>roads_map <span class="sc">+</span> <span class="fu">tm_shape</span>(railways) <span class="sc">+</span> <span class="fu">tm_lines</span>(<span class="at">col =</span> <span class="st">&quot;black&quot;</span>, <span class="at">lwd =</span> <span class="dv">2</span>, <span class="at">lty =</span> <span class="st">&quot;solid&quot;</span>)</span></code></pre></div>
<p><img src="r5_files/figure-html/unnamed-chunk-25-1.png" width="672" /></p>
<p>Next, drop all attribute fields, create a new <code>SCORE</code>
field for each layer, and populate with the value zero:</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" tabindex="-1"></a><span class="co"># Create new sf objects that retain only the geometry column</span></span>
<span id="cb32-2"><a href="#cb32-2" tabindex="-1"></a><span class="co"># This effectively drops all other attribute fields</span></span>
<span id="cb32-3"><a href="#cb32-3" tabindex="-1"></a>built_areas_clean <span class="ot">&lt;-</span> <span class="fu">st_sf</span>(<span class="at">geometry =</span> <span class="fu">st_geometry</span>(built_areas))</span>
<span id="cb32-4"><a href="#cb32-4" tabindex="-1"></a>roads_clean <span class="ot">&lt;-</span> <span class="fu">st_sf</span>(<span class="at">geometry =</span> <span class="fu">st_geometry</span>(roads_filtered))</span>
<span id="cb32-5"><a href="#cb32-5" tabindex="-1"></a>railways_clean <span class="ot">&lt;-</span> <span class="fu">st_sf</span>(<span class="at">geometry =</span> <span class="fu">st_geometry</span>(railways))</span>
<span id="cb32-6"><a href="#cb32-6" tabindex="-1"></a></span>
<span id="cb32-7"><a href="#cb32-7" tabindex="-1"></a><span class="co"># Add a new attribute field called SCORE and populate it with 0 for each feature</span></span>
<span id="cb32-8"><a href="#cb32-8" tabindex="-1"></a>built_areas_clean<span class="sc">$</span>SCORE <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb32-9"><a href="#cb32-9" tabindex="-1"></a>roads_clean<span class="sc">$</span>SCORE <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb32-10"><a href="#cb32-10" tabindex="-1"></a>railways_clean<span class="sc">$</span>SCORE <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb32-11"><a href="#cb32-11" tabindex="-1"></a></span>
<span id="cb32-12"><a href="#cb32-12" tabindex="-1"></a><span class="co"># Optionally, reorder the columns so that SCORE comes before the geometry column</span></span>
<span id="cb32-13"><a href="#cb32-13" tabindex="-1"></a>built_areas_clean <span class="ot">&lt;-</span> built_areas_clean[, <span class="fu">c</span>(<span class="st">&quot;SCORE&quot;</span>, <span class="st">&quot;geometry&quot;</span>)]</span>
<span id="cb32-14"><a href="#cb32-14" tabindex="-1"></a>roads_clean <span class="ot">&lt;-</span> roads_clean[, <span class="fu">c</span>(<span class="st">&quot;SCORE&quot;</span>, <span class="st">&quot;geometry&quot;</span>)]</span>
<span id="cb32-15"><a href="#cb32-15" tabindex="-1"></a>railways_clean <span class="ot">&lt;-</span> railways_clean[, <span class="fu">c</span>(<span class="st">&quot;SCORE&quot;</span>, <span class="st">&quot;geometry&quot;</span>)]</span></code></pre></div>
<p>Create the 300 m buffer and dissolve features:</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" tabindex="-1"></a><span class="co"># Create a 300m buffer around each point</span></span>
<span id="cb33-2"><a href="#cb33-2" tabindex="-1"></a>built_buffers <span class="ot">&lt;-</span> <span class="fu">st_buffer</span>(built_areas_clean, <span class="at">dist =</span> <span class="dv">300</span>)</span>
<span id="cb33-3"><a href="#cb33-3" tabindex="-1"></a>roads_buffers <span class="ot">&lt;-</span> <span class="fu">st_buffer</span>(roads_clean, <span class="at">dist =</span> <span class="dv">300</span>)</span>
<span id="cb33-4"><a href="#cb33-4" tabindex="-1"></a>railways_buffers <span class="ot">&lt;-</span> <span class="fu">st_buffer</span>(railways_clean, <span class="at">dist =</span> <span class="dv">300</span>)</span>
<span id="cb33-5"><a href="#cb33-5" tabindex="-1"></a></span>
<span id="cb33-6"><a href="#cb33-6" tabindex="-1"></a><span class="co"># Dissolve (union) buffers by the SCORE field</span></span>
<span id="cb33-7"><a href="#cb33-7" tabindex="-1"></a>built_buffers_dissolved <span class="ot">&lt;-</span> built_buffers <span class="sc">%&gt;%</span></span>
<span id="cb33-8"><a href="#cb33-8" tabindex="-1"></a>  <span class="fu">group_by</span>(SCORE) <span class="sc">%&gt;%</span></span>
<span id="cb33-9"><a href="#cb33-9" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">geometry =</span> <span class="fu">st_union</span>(geometry))</span>
<span id="cb33-10"><a href="#cb33-10" tabindex="-1"></a>roads_buffers_dissolved <span class="ot">&lt;-</span> roads_buffers <span class="sc">%&gt;%</span></span>
<span id="cb33-11"><a href="#cb33-11" tabindex="-1"></a>  <span class="fu">group_by</span>(SCORE) <span class="sc">%&gt;%</span></span>
<span id="cb33-12"><a href="#cb33-12" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">geometry =</span> <span class="fu">st_union</span>(geometry))</span>
<span id="cb33-13"><a href="#cb33-13" tabindex="-1"></a>railways_buffers_dissolved <span class="ot">&lt;-</span> railways_buffers <span class="sc">%&gt;%</span></span>
<span id="cb33-14"><a href="#cb33-14" tabindex="-1"></a>  <span class="fu">group_by</span>(SCORE) <span class="sc">%&gt;%</span></span>
<span id="cb33-15"><a href="#cb33-15" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">geometry =</span> <span class="fu">st_union</span>(geometry))</span>
<span id="cb33-16"><a href="#cb33-16" tabindex="-1"></a></span>
<span id="cb33-17"><a href="#cb33-17" tabindex="-1"></a><span class="co"># Join the three layers together and dissolve again</span></span>
<span id="cb33-18"><a href="#cb33-18" tabindex="-1"></a><span class="co"># Combine the sf objects using rbind (works if they have identical column names)</span></span>
<span id="cb33-19"><a href="#cb33-19" tabindex="-1"></a>infrastructure <span class="ot">&lt;-</span> <span class="fu">rbind</span>(built_buffers_dissolved,</span>
<span id="cb33-20"><a href="#cb33-20" tabindex="-1"></a>                        roads_buffers_dissolved,</span>
<span id="cb33-21"><a href="#cb33-21" tabindex="-1"></a>                        railways_buffers_dissolved</span>
<span id="cb33-22"><a href="#cb33-22" tabindex="-1"></a>                        )</span>
<span id="cb33-23"><a href="#cb33-23" tabindex="-1"></a></span>
<span id="cb33-24"><a href="#cb33-24" tabindex="-1"></a><span class="co"># Dissolve (union)</span></span>
<span id="cb33-25"><a href="#cb33-25" tabindex="-1"></a>infrastructure_dissolved <span class="ot">&lt;-</span> <span class="fu">st_union</span>(infrastructure)</span>
<span id="cb33-26"><a href="#cb33-26" tabindex="-1"></a></span>
<span id="cb33-27"><a href="#cb33-27" tabindex="-1"></a><span class="co"># Convert the unioned geometry back into an sf object</span></span>
<span id="cb33-28"><a href="#cb33-28" tabindex="-1"></a>infrastructure_sf <span class="ot">&lt;-</span> <span class="fu">st_sf</span>(<span class="at">geometry =</span> infrastructure_dissolved)</span>
<span id="cb33-29"><a href="#cb33-29" tabindex="-1"></a></span>
<span id="cb33-30"><a href="#cb33-30" tabindex="-1"></a><span class="co"># Add SCORE field and sort attribute columns</span></span>
<span id="cb33-31"><a href="#cb33-31" tabindex="-1"></a>infrastructure_sf<span class="sc">$</span>SCORE <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb33-32"><a href="#cb33-32" tabindex="-1"></a>infrastructure_sf <span class="ot">&lt;-</span> infrastructure_sf[, <span class="fu">c</span>(<span class="st">&quot;SCORE&quot;</span>, <span class="st">&quot;geometry&quot;</span>)]</span>
<span id="cb33-33"><a href="#cb33-33" tabindex="-1"></a></span>
<span id="cb33-34"><a href="#cb33-34" tabindex="-1"></a><span class="co"># Plot bnd</span></span>
<span id="cb33-35"><a href="#cb33-35" tabindex="-1"></a>bnd_map <span class="ot">&lt;-</span> <span class="fu">tm_shape</span>(hunter_bnd) <span class="sc">+</span>  <span class="fu">tm_borders</span>(<span class="st">&quot;black&quot;</span>) <span class="sc">+</span> <span class="fu">tm_layout</span>(<span class="at">frame =</span> <span class="cn">FALSE</span>)</span>
<span id="cb33-36"><a href="#cb33-36" tabindex="-1"></a></span>
<span id="cb33-37"><a href="#cb33-37" tabindex="-1"></a><span class="co"># Add infrastructure</span></span>
<span id="cb33-38"><a href="#cb33-38" tabindex="-1"></a>bnd_map <span class="sc">+</span> <span class="fu">tm_shape</span>(infrastructure_sf) <span class="sc">+</span> </span>
<span id="cb33-39"><a href="#cb33-39" tabindex="-1"></a>        <span class="fu">tm_fill</span>(<span class="st">&quot;gray&quot;</span>) <span class="sc">+</span> </span>
<span id="cb33-40"><a href="#cb33-40" tabindex="-1"></a>        <span class="fu">tm_layout</span>(<span class="at">frame =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p><img src="r5_files/figure-html/unnamed-chunk-27-1.png" width="80%" style="display: block; margin: auto;" /></p>
<p>We are now ready to convert the infrastructure buffer layer
(<code>infrastructure_dissolved</code>) to a raster with scores:</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" tabindex="-1"></a><span class="co"># Convert the sf object to a terra SpatVector</span></span>
<span id="cb34-2"><a href="#cb34-2" tabindex="-1"></a>infrastructure_terra <span class="ot">&lt;-</span> <span class="fu">vect</span>(infrastructure_sf)</span>
<span id="cb34-3"><a href="#cb34-3" tabindex="-1"></a></span>
<span id="cb34-4"><a href="#cb34-4" tabindex="-1"></a><span class="co"># Read in the target SpatRaster that defines the extent and resolution</span></span>
<span id="cb34-5"><a href="#cb34-5" tabindex="-1"></a><span class="co"># Here we read in one of the Eco Logical GeoTIFFs provided</span></span>
<span id="cb34-6"><a href="#cb34-6" tabindex="-1"></a>target_raster <span class="ot">&lt;-</span> <span class="fu">rast</span>(<span class="st">&quot;./data/data_koala/raster/allveg_patch_size.tif&quot;</span>)</span>
<span id="cb34-7"><a href="#cb34-7" tabindex="-1"></a></span>
<span id="cb34-8"><a href="#cb34-8" tabindex="-1"></a><span class="co"># Rasterise the polygons using an attribute field</span></span>
<span id="cb34-9"><a href="#cb34-9" tabindex="-1"></a>infrastructure_rast <span class="ot">&lt;-</span> <span class="fu">rasterize</span>(infrastructure_terra, target_raster, <span class="at">field =</span> <span class="st">&quot;SCORE&quot;</span>)</span>
<span id="cb34-10"><a href="#cb34-10" tabindex="-1"></a></span>
<span id="cb34-11"><a href="#cb34-11" tabindex="-1"></a><span class="co"># Convert all NA (no data) values in the new raster to 300</span></span>
<span id="cb34-12"><a href="#cb34-12" tabindex="-1"></a>infrastructure_rast[<span class="fu">is.na</span>(infrastructure_rast)] <span class="ot">&lt;-</span> <span class="dv">300</span></span>
<span id="cb34-13"><a href="#cb34-13" tabindex="-1"></a></span>
<span id="cb34-14"><a href="#cb34-14" tabindex="-1"></a><span class="co"># Plot data for quick visual inspection</span></span>
<span id="cb34-15"><a href="#cb34-15" tabindex="-1"></a>infra_map <span class="ot">&lt;-</span> <span class="fu">tm_shape</span>(infrastructure_rast) <span class="sc">+</span> </span>
<span id="cb34-16"><a href="#cb34-16" tabindex="-1"></a>       <span class="fu">tm_raster</span>(<span class="at">col.scale =</span> <span class="fu">tm_scale_categorical</span>()) <span class="sc">+</span> </span>
<span id="cb34-17"><a href="#cb34-17" tabindex="-1"></a>       <span class="fu">tm_layout</span>(<span class="at">frame =</span> <span class="cn">FALSE</span>, <span class="at">legend.text.size =</span> <span class="fl">0.6</span>, <span class="at">legend.title.size =</span> <span class="fl">0.6</span>)</span>
<span id="cb34-18"><a href="#cb34-18" tabindex="-1"></a></span>
<span id="cb34-19"><a href="#cb34-19" tabindex="-1"></a><span class="co"># Plot study area bnd on top</span></span>
<span id="cb34-20"><a href="#cb34-20" tabindex="-1"></a>infra_map <span class="sc">+</span> <span class="fu">tm_shape</span>(hunter_bnd) <span class="sc">+</span>  <span class="fu">tm_borders</span>(<span class="st">&quot;white&quot;</span>) </span></code></pre></div>
<pre><code>## SpatRaster object downsampled to 3018 by 3314 cells.</code></pre>
<p><img src="r5_files/figure-html/unnamed-chunk-28-1.png" width="80%" style="display: block; margin: auto;" /></p>
<p>We have now calculated the three missing final layers:</p>
<ul>
<li><span class="math inline">\(soil\)</span> : soil fertility. This is
layer <code>soil_landscapes_rast</code></li>
<li><span class="math inline">\(koala\)</span> : buffered koala
sightings. This is layer <code>koala_buffers_rast</code></li>
<li><span class="math inline">\(urb\)</span> : distance to
infrastructure. This is layer <code>infrastructure_rast</code></li>
</ul>
</div>
</div>
<div id="calculate-koala-habitat-value" class="section level1">
<h1>Calculate koala habitat value</h1>
<p>To repeat from above, the final koala habitat suitability map is
calculated by combining all the scores assigned to each data layer, with
each layer further multiplied by a weight:</p>
<p>The calculation is as follows:</p>
<p><span class="math display">\[khv = [(veg_{prim} \times 3) +
(veg_{sec} \times 2) + veg_{supp}] + (soil \times 2) +\]</span> <span
class="math display">\[+ (veg_{water} \times 2) + koala + (urb \times 3)
+ (veg_{patch} \times 3)\]</span> where:</p>
<ul>
<li><span class="math inline">\(veg_{prim}\)</span> : primary feed
trees</li>
<li><span class="math inline">\(veg_{sec}\)</span> : secondary feed
trees</li>
<li><span class="math inline">\(veg_{supp}\)</span> : supplementary
trees</li>
<li><span class="math inline">\(soil\)</span> : soil fertility</li>
<li><span class="math inline">\(veg_{water}\)</span> : vegetation
proximity to water</li>
<li><span class="math inline">\(koala\)</span> : buffered koala
sightings</li>
<li><span class="math inline">\(urb\)</span> : distance to
infrastructure</li>
<li><span class="math inline">\(veg_{patch}\)</span> : vegetation patch
size</li>
</ul>
<p>Calculating <span class="math inline">\(khv\)</span> using map
algebra is straight forward:</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" tabindex="-1"></a><span class="co"># Read in Eco Logical rasters</span></span>
<span id="cb36-2"><a href="#cb36-2" tabindex="-1"></a>veg_prim <span class="ot">&lt;-</span> <span class="fu">rast</span>(<span class="st">&quot;./data/data_koala/raster/allveg_prim.tif&quot;</span>)</span>
<span id="cb36-3"><a href="#cb36-3" tabindex="-1"></a>veg_sec <span class="ot">&lt;-</span> <span class="fu">rast</span>(<span class="st">&quot;./data/data_koala/raster/allveg_sec.tif&quot;</span>)</span>
<span id="cb36-4"><a href="#cb36-4" tabindex="-1"></a>veg_supp <span class="ot">&lt;-</span> <span class="fu">rast</span>(<span class="st">&quot;./data/data_koala/raster/allveg_supp.tif&quot;</span>)</span>
<span id="cb36-5"><a href="#cb36-5" tabindex="-1"></a>veg_water <span class="ot">&lt;-</span> <span class="fu">rast</span>(<span class="st">&quot;./data/data_koala/raster/dist_to_water.tif&quot;</span>)</span>
<span id="cb36-6"><a href="#cb36-6" tabindex="-1"></a>veg_patch <span class="ot">&lt;-</span> <span class="fu">rast</span>(<span class="st">&quot;./data/data_koala/raster/allveg_patch_size.tif&quot;</span>)</span>
<span id="cb36-7"><a href="#cb36-7" tabindex="-1"></a></span>
<span id="cb36-8"><a href="#cb36-8" tabindex="-1"></a><span class="co"># Map algebra</span></span>
<span id="cb36-9"><a href="#cb36-9" tabindex="-1"></a>koala_habitat <span class="ot">&lt;-</span> (<span class="dv">3</span> <span class="sc">*</span> veg_prim) <span class="sc">+</span> (<span class="dv">2</span> <span class="sc">*</span> veg_sec) <span class="sc">+</span> veg_supp <span class="sc">+</span> </span>
<span id="cb36-10"><a href="#cb36-10" tabindex="-1"></a>                 (<span class="dv">2</span> <span class="sc">*</span> soil_landscapes_rast) <span class="sc">+</span> </span>
<span id="cb36-11"><a href="#cb36-11" tabindex="-1"></a>                 (<span class="dv">2</span> <span class="sc">*</span> veg_water) <span class="sc">+</span> </span>
<span id="cb36-12"><a href="#cb36-12" tabindex="-1"></a>                 koala_buffers_rast <span class="sc">+</span> </span>
<span id="cb36-13"><a href="#cb36-13" tabindex="-1"></a>                 (<span class="dv">3</span> <span class="sc">*</span> infrastructure_rast) <span class="sc">+</span> </span>
<span id="cb36-14"><a href="#cb36-14" tabindex="-1"></a>                 (<span class="dv">3</span> <span class="sc">*</span> veg_patch)</span>
<span id="cb36-15"><a href="#cb36-15" tabindex="-1"></a></span>
<span id="cb36-16"><a href="#cb36-16" tabindex="-1"></a><span class="co">#Change the name of the layer to &quot;khv&quot;</span></span>
<span id="cb36-17"><a href="#cb36-17" tabindex="-1"></a><span class="fu">names</span>(koala_habitat) <span class="ot">&lt;-</span> <span class="st">&quot;khv&quot;</span></span>
<span id="cb36-18"><a href="#cb36-18" tabindex="-1"></a></span>
<span id="cb36-19"><a href="#cb36-19" tabindex="-1"></a><span class="co"># Plot bnd</span></span>
<span id="cb36-20"><a href="#cb36-20" tabindex="-1"></a>khv_map <span class="ot">&lt;-</span> <span class="fu">tm_shape</span>(hunter_bnd) <span class="sc">+</span>  <span class="fu">tm_borders</span>(<span class="st">&quot;black&quot;</span>) </span>
<span id="cb36-21"><a href="#cb36-21" tabindex="-1"></a></span>
<span id="cb36-22"><a href="#cb36-22" tabindex="-1"></a><span class="co"># Add khv</span></span>
<span id="cb36-23"><a href="#cb36-23" tabindex="-1"></a>khv_map <span class="sc">+</span> <span class="fu">tm_shape</span>(koala_habitat) <span class="sc">+</span> </span>
<span id="cb36-24"><a href="#cb36-24" tabindex="-1"></a>    <span class="fu">tm_raster</span>(<span class="st">&quot;khv&quot;</span>, </span>
<span id="cb36-25"><a href="#cb36-25" tabindex="-1"></a>            <span class="at">col.scale =</span> <span class="fu">tm_scale_continuous_sqrt</span>(</span>
<span id="cb36-26"><a href="#cb36-26" tabindex="-1"></a>              <span class="at">values =</span> <span class="st">&quot;viridis&quot;</span>, </span>
<span id="cb36-27"><a href="#cb36-27" tabindex="-1"></a>              ), </span>
<span id="cb36-28"><a href="#cb36-28" tabindex="-1"></a>            <span class="at">col.legend =</span> <span class="fu">tm_legend</span>(</span>
<span id="cb36-29"><a href="#cb36-29" tabindex="-1"></a>              <span class="at">title =</span> <span class="st">&quot;Habitat </span><span class="sc">\n</span><span class="st">Suitability </span><span class="sc">\n</span><span class="st">Index&quot;</span>,</span>
<span id="cb36-30"><a href="#cb36-30" tabindex="-1"></a>              <span class="at">position =</span> <span class="fu">tm_pos_out</span>(<span class="at">cell.h =</span> <span class="st">&quot;right&quot;</span>, <span class="at">cell.v =</span> <span class="st">&quot;center&quot;</span>),</span>
<span id="cb36-31"><a href="#cb36-31" tabindex="-1"></a>              <span class="at">orientation =</span> <span class="st">&quot;portrait&quot;</span></span>
<span id="cb36-32"><a href="#cb36-32" tabindex="-1"></a>              )</span>
<span id="cb36-33"><a href="#cb36-33" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb36-34"><a href="#cb36-34" tabindex="-1"></a>    <span class="fu">tm_layout</span>(<span class="at">frame =</span> <span class="cn">FALSE</span>, <span class="at">legend.text.size =</span> <span class="fl">0.6</span>, <span class="at">legend.title.size =</span> <span class="fl">0.6</span>) </span></code></pre></div>
<pre><code>## SpatRaster object downsampled to 3018 by 3314 cells.</code></pre>
<p><img src="r5_files/figure-html/unnamed-chunk-29-1.png" width="80%" style="display: block; margin: auto;" /></p>
<p>To calculate the normalised <span class="math inline">\(khv\)</span>,
first we check the maximum value:</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" tabindex="-1"></a><span class="fu">print</span>(koala_habitat)</span></code></pre></div>
<pre><code>## class       : SpatRaster 
## dimensions  : 3793, 4165, 1  (nrow, ncol, nlyr)
## resolution  : 25, 25  (x, y)
## extent      : 332811.5, 436936.5, 6312635, 6407460  (xmin, xmax, ymin, ymax)
## coord. ref. : GDA_1994_Transverse_Mercator 
## source(s)   : memory
## varname     : allveg_prim 
## name        :  khv 
## min value   :    0 
## max value   : 3000</code></pre>
<p>The maximum <span class="math inline">\(khv\)</span> value is 3000,
and so to normalise the <span class="math inline">\(khv\)</span> values,
we divide all grid cells of the <code>koala_habitat</code> raster with
3000 and then multiply by 100:</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" tabindex="-1"></a>koala_habitat_norm <span class="ot">&lt;-</span> (koala_habitat <span class="sc">/</span> <span class="dv">3000</span>) <span class="sc">*</span> <span class="dv">100</span></span></code></pre></div>
<p>We are now ready to make our final map:</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" tabindex="-1"></a><span class="co"># Draw bnd</span></span>
<span id="cb41-2"><a href="#cb41-2" tabindex="-1"></a>khvnorm_map <span class="ot">&lt;-</span> <span class="fu">tm_shape</span>(hunter_bnd) <span class="sc">+</span>  <span class="fu">tm_borders</span>(<span class="st">&quot;black&quot;</span>) </span>
<span id="cb41-3"><a href="#cb41-3" tabindex="-1"></a></span>
<span id="cb41-4"><a href="#cb41-4" tabindex="-1"></a><span class="co"># Add normalised khv  </span></span>
<span id="cb41-5"><a href="#cb41-5" tabindex="-1"></a>khvnorm_map <span class="sc">+</span> <span class="fu">tm_shape</span>(koala_habitat_norm) <span class="sc">+</span> </span>
<span id="cb41-6"><a href="#cb41-6" tabindex="-1"></a>       <span class="fu">tm_raster</span>(<span class="st">&quot;khv&quot;</span>,</span>
<span id="cb41-7"><a href="#cb41-7" tabindex="-1"></a>          <span class="at">col.scale =</span> <span class="fu">tm_scale_intervals</span>(                               <span class="co"># Custom breaks</span></span>
<span id="cb41-8"><a href="#cb41-8" tabindex="-1"></a>            <span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;#584053&quot;</span>, <span class="st">&quot;#8dc6bf&quot;</span>, <span class="st">&quot;#fcbc66&quot;</span>, <span class="st">&quot;#f97b4f&quot;</span>),     <span class="co"># Define colours</span></span>
<span id="cb41-9"><a href="#cb41-9" tabindex="-1"></a>            <span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">20</span>, <span class="dv">34</span>, <span class="dv">49</span>, <span class="dv">100</span>),           </span>
<span id="cb41-10"><a href="#cb41-10" tabindex="-1"></a>            <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;low&quot;</span>, <span class="st">&quot;moderate&quot;</span>, <span class="st">&quot;high&quot;</span>, <span class="st">&quot;very high&quot;</span>)          <span class="co"># Legend labels</span></span>
<span id="cb41-11"><a href="#cb41-11" tabindex="-1"></a>            ), </span>
<span id="cb41-12"><a href="#cb41-12" tabindex="-1"></a>          <span class="at">col.legend =</span> <span class="fu">tm_legend</span>(</span>
<span id="cb41-13"><a href="#cb41-13" tabindex="-1"></a>            <span class="at">title =</span> <span class="st">&quot;HSV Class&quot;</span>,</span>
<span id="cb41-14"><a href="#cb41-14" tabindex="-1"></a>            <span class="at">position =</span> <span class="fu">tm_pos_out</span>(<span class="at">cell.h =</span> <span class="st">&quot;right&quot;</span>, <span class="at">cell.v =</span> <span class="st">&quot;center&quot;</span>),</span>
<span id="cb41-15"><a href="#cb41-15" tabindex="-1"></a>            <span class="at">orientation =</span> <span class="st">&quot;portrait&quot;</span></span>
<span id="cb41-16"><a href="#cb41-16" tabindex="-1"></a>            )</span>
<span id="cb41-17"><a href="#cb41-17" tabindex="-1"></a>       )<span class="sc">+</span></span>
<span id="cb41-18"><a href="#cb41-18" tabindex="-1"></a>       <span class="fu">tm_layout</span>(<span class="at">frame =</span> <span class="cn">FALSE</span>, <span class="at">legend.text.size =</span> <span class="fl">0.6</span>, <span class="at">legend.title.size =</span> <span class="fl">0.6</span>)</span></code></pre></div>
<pre><code>## SpatRaster object downsampled to 3018 by 3314 cells.</code></pre>
<p><img src="r5_files/figure-html/unnamed-chunk-32-1.png" width="80%" style="display: block; margin: auto;" /></p>
</div>
<div id="summary" class="section level1">
<h1>Summary</h1>
<p>The analyses above replicated a koala population sustainability study
conducted in the Lower Hunter region of NSW for the <em>Department of
Sustainability, Environment, Water, Population and Communities</em> by
the consultancy <em>Eco Logical</em>. While some data sources differed
from those used in the original study, the resulting koala habitat
suitability maps closely align with the original findings, highlighting
the same areas of high and very high suitability.</p>
</div>
<div id="list-of-r-functions" class="section level1">
<h1>List of R functions</h1>
<p>This practical showcased a diverse range of R functions. A summary of
these is presented below:</p>
<p><strong>base R functions</strong></p>
<ul>
<li><p><code>c()</code>: concatenates its arguments to create a vector,
which is one of the most fundamental data structures in R. It is used to
combine numbers, strings, and other objects into a single
vector.</p></li>
<li><p><code>head()</code>: returns the first few elements or rows of an
object, such as a vector, matrix, or data frame. It is a convenient tool
for quickly inspecting the structure and content of data.</p></li>
<li><p><code>na</code>: represents missing or undefined values in R. It
is integral to data analysis as it denotes the absence of a value and
requires careful handling in computations and data
manipulations.</p></li>
<li><p><code>names()</code>: retrieves or sets the names attribute of an
R object, such as a vector, list, or data frame. It is essential for
labeling elements to improve readability and facilitate data
management.</p></li>
<li><p><code>print()</code>: outputs the value of an R object to the
console. It is particularly useful for debugging or for ensuring that
outputs are displayed correctly within functions.</p></li>
<li><p><code>rbind()</code>: combines R objects by rows, stacking them
vertically to form a new matrix or data frame. It is especially useful
for merging datasets that share the same column structure.</p></li>
<li><p><code>unique()</code>: extracts distinct elements from a vector,
data frame, or other R object by removing duplicate entries. It
preserves the order of first occurrences and is useful for identifying
unique values in a dataset.</p></li>
</ul>
<p><strong>dplyr (tidyverse) functions</strong></p>
<ul>
<li><p><code>case_when()</code>: acts as a vectorized if-else statement
by evaluating multiple conditions and returning corresponding values. It
is useful for creating new variables based on complex conditional
logic.</p></li>
<li><p><code>filter()</code>: subsets rows of a data frame based on
specified logical conditions. It is essential for extracting relevant
data for analysis.</p></li>
<li><p><code>group_by()</code>: organizes data into groups based on one
or more variables, setting the stage for group-wise operations. It is
often used together with summarise or mutate.</p></li>
<li><p><code>mutate()</code>: adds new columns or transforms existing
ones within a data frame. It is essential for creating derived variables
and modifying data within the tidyverse workflow.</p></li>
<li><p><code>rename()</code>: changes the names of columns in a data
frame to make them more descriptive or consistent. It provides a clear
syntax for renaming variables during data cleaning.</p></li>
<li><p><code>select()</code>: extracts a subset of columns from a data
frame based on specified criteria. It simplifies data frames by keeping
only the variables needed for analysis.</p></li>
<li><p><code>summarise()</code>: reduces multiple values down to a
single summary statistic per group in a data frame. When used with
<code>group_by()</code>, it facilitates the computation of aggregates
such as mean, sum, or count.</p></li>
</ul>
<p><strong>sf functions</strong></p>
<ul>
<li><p><code>read_sf()</code>: reads spatial data from various file
formats into an sf object. It simplifies the import of geographic data
for spatial analysis.</p></li>
<li><p><code>st_buffer()</code>: creates a buffer around spatial
features at a specified distance. It is useful for proximity analyses
and for creating zones of influence.</p></li>
<li><p><code>st_geometry()</code>: extracts or assigns the geometry
component of an sf object. It allows you to isolate and manipulate the
spatial aspects of the data separately from the attributes.</p></li>
<li><p><code>st_intersection()</code>: computes the geometric
intersection between spatial features. It is used to identify
overlapping areas or shared boundaries between different spatial
objects.</p></li>
<li><p><code>st_sf()</code>: converts a regular data frame into an sf
object by attaching spatial geometry information. It is the foundational
step for performing spatial analysis using the simple features
framework.</p></li>
<li><p><code>st_union()</code>: merges multiple spatial features into a
single combined geometry. It is useful for dissolving boundaries and
creating aggregate spatial features.</p></li>
</ul>
<p><strong>terra functions</strong></p>
<ul>
<li><p><code>rast()</code>: creates a raster object from files,
matrices, or arrays, and is used for spatial analyses and modeling. It
is the entry point for working with raster data in terra.</p></li>
<li><p><code>rasterize()</code>: converts vector data (points, lines, or
polygons) into a raster format. It is key for aligning vector data with
a raster grid for subsequent analysis.</p></li>
<li><p><code>vect()</code>: reads and creates vector objects from
various spatial data formats. It handles points, lines, and polygons for
versatile spatial data manipulation.</p></li>
</ul>
<p><strong>tmap functions</strong></p>
<ul>
<li><p><code>tm_shape()</code>: initializes a tmap object with a spatial
dataset, setting up the base layer for thematic maps. It establishes the
spatial context for subsequent map layers.</p></li>
<li><p><code>tm_borders()</code>: adds border lines to spatial objects
on a map, highlighting boundaries. It enhances map readability by
clearly delineating regions.</p></li>
<li><p><code>tm_fill()</code>: fills polygon areas with specified colors
based on attribute data, providing visual context to thematic maps. It
is commonly used to represent different categories or
intensities.</p></li>
<li><p><code>tm_dots()</code>: plots point features on a tmap, useful
for visualizing locations or events. It supports customization of size,
color, and shape to enhance map interpretation.</p></li>
<li><p><code>tm_lines()</code>: draws line features on a map, ideal for
representing roads, rivers, or connections. It provides options for
styling, such as line width and color.</p></li>
<li><p><code>tm_raster()</code>: displays raster data layers on a map,
allowing integration of continuous data like elevation or temperature.
It supports various styling and transparency options.</p></li>
<li><p><code>tm_layout()</code>: customizes the overall appearance and
layout of the map, including margins, titles, and backgrounds. It helps
fine-tune the visual presentation of the map.</p></li>
<li><p><code>tm_legend()</code>: adjusts the properties and placement of
the map legend, improving clarity and context. It offers options to
modify text size, symbols, and arrangement.</p></li>
<li><p><code>tm_pos_out()</code>: positions map elements such as legends
or scale bars outside the main map panel. It helps reduce clutter within
the map area.</p></li>
<li><p><code>tm_scale_categorical()</code>: adds a categorical scale bar
to represent discrete variable classes on the map. It is tailored for
visualizing grouped data.</p></li>
<li><p><code>tm_scale_continuous_sqrt()</code>: applies a square root
transformation to a continuous color scale in tmap. It is useful for
visualizing data with wide-ranging values by reducing skew.</p></li>
<li><p><code>tm_scale_intervals()</code>: creates scale intervals to
categorize continuous data into bins. It enhances interpretability by
breaking down data into meaningful segments.</p></li>
</ul>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
